android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -469,6 +469,10 @@ protected static void deleteLexicalModel(Context context, int position, boolean
   @SuppressWarnings("unchecked")
   private static ArrayList<HashMap<String, String>> getList(Context context, String filename) {
     ArrayList<HashMap<String, String>> list = null;
+    if (context == null) {
```

Comment:
I'd prefer to make this test in `onKeyboardDownloadFinished` against `currentContext`. There's even a function `isContextAvailable()` which might help -- although it needs tweaking...

I see the only place where `currentContext` gets set is in `checkForResourceUpdates()` (ResourcesUpdateTool.java:448). It would be good to understand if there is a code path that could result in a `null` `currentContext`.

Reply:
> I'd prefer to make this test in `onKeyboardDownloadFinished` against `currentContext`. There's even a function `isContextAvailable()` which might help -- although it needs tweaking...

To clarify, are you wanting the test in `ResourceUpdateTool.onKeyboardDownloadFinished()`?

Reply:
For 14.0, `onKeyboardDownloadFinished()` does nothing, so I'll move the checks to `onPackageInstalled()` and `onLexicalModelInstalled()`